Yosemite/Yosemite/Stores/Order/OrderSyncProductInput.swift

```diff
@@ -27,24 +38,26 @@ public struct OrderSyncProductInput {
     /// Types of products the synchronizer supports
     ///
     public enum ProductType: Hashable {
-        case product(OrderSyncProductTypeProtocol)
+        case product(any OrderSyncProductTypeProtocol)
         case variation(ProductVariation)
 
         public func hash(into hasher: inout Hasher) {
             switch self {
             case .product(let product):
-                hasher.combine("product-\(product.productID)")
+                hasher.combine("productType-product-")
```

Comment:
super nit: feels like there's something after the last dash, maybe can remove it like in the variation case?

```suggestion
                hasher.combine("productType-product")
```

Reply:
Thanks, good spot (though I dread having to get this through CI again 🙄